crates/rpc/rpc-eth-api/src/helpers/pending_block.rs

```diff
@@ -322,10 +322,47 @@ pub trait LoadPendingBlock:
             cumulative_gas_used += gas_used;
         }
 
-        let BlockBuilderOutcome { execution_result, block, .. } =
+        let BlockBuilderOutcome { execution_result, block, hashed_state, .. } =
             builder.finish(&state_provider).map_err(Self::Error::from_eth_err)?;
 
-        Ok((block, execution_result.receipts))
+        Ok(ExecutedBlock {
+            recovered_block: block.into(),
+            execution_output: Arc::new(ExecutionOutcome {
+                receipts: vec![execution_result.receipts],
+                ..Default::default()
+            }),
+            hashed_state: Arc::new(hashed_state),
+        })
+    }
+
+    /// Returns the pending state for the given block id.
+    fn get_pending_state(
+        &self,
+        id: &BlockId,
+    ) -> impl std::future::Future<Output = Result<StateProviderBox, Self::Error>> + Send {
+        async move {
```

Comment:
what this should do is, is first obtain the latest state

and then use 

https://github.com/paradigmxyz/reth/blob/876e964cbcbd85ade6fdec40bb35f08690332854/crates/chain-state/src/memory_overlay.rs#L36-L46

if we have a pending block, so that the pending state extends the latest state